### PR TITLE
Add AWSQueryCompatible trait support

### PR DESCRIPTION
**Pull request by eddy-aws**
The awsQueryCompatible error code is returned if the error header is found in the response. This is a pre-requisite for migrating services from AWSQuery wire protocol to AWSJson.

### CODE REVIEW

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsProtocolUtils.java [hunk 1]
```diff
@@ -223,6 +223,20 @@ static void generateHttpProtocolTests(GenerationContext context) {
         ).generateProtocolTests();
     }
 
+    public static void writeAwsQueryErrorCodeDeserializer(GenerationContext context) {
+        GoWriter writer = context.getWriter().get();
+        writer.write("queryCodeHeader := response.Header.Get(\"x-amzn-query-error\")");
+        writer.write("if queryCodeHeader != \"\" {");
```
Comment (kstich) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsProtocolUtils.java:229]: `if` statements can use the `openBlock` method on the writer to automatically handle indentation.

File: .changelog/41575353444b40ffbf474f4155544f00.json [hunk 2]
```diff
@@ -0,0 +1,331 @@
+{
```
Comment (syall) [inline .changelog/41575353444b40ffbf474f4155544f00.json:1]: This changelog entry is generated incorrectly.

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java [hunk 3]
```diff
@@ -174,6 +176,9 @@ public void generateProtocolTests(GenerationContext context) {
     @Override
     protected void writeErrorMessageCodeDeserializer(GenerationContext context) {
         writeJsonErrorMessageCodeDeserializer(context);
+        if (context.getService().hasTrait(AwsQueryCompatibleTrait.class)) {
+            writeAwsQueryErrorCodeDeserializer(context);
+        }
     }
```
Comment (syall) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java:182]: From what I understand, this trait / behavior is only for the [AWS JSON 1.0 protocol](https://smithy.io/2.0/aws/protocols/aws-json-1_0-protocol.html).

If that's the case, the existing code in the generalized `JsonRpcProtocolGenerator` should not be touched, and instead be updated in [`AwsJsonRpc1_0`](https://github.com/aws/aws-sdk-go-v2/blob/main/codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsJsonRpc1_0.java) with a method override.

File: codegen/gradle.properties [hunk 4]
```diff
@@ -1,2 +1,2 @@
-smithyVersion=1.25.0
```
Comment (syall) [inline codegen/gradle.properties:1]: Split the smithy upgrade to a separate commit and with its own changelog entry.

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java [hunk 5]
```diff
@@ -42,6 +43,7 @@
 import software.amazon.smithy.model.traits.ErrorTrait;
 import software.amazon.smithy.model.traits.EventHeaderTrait;
 import software.amazon.smithy.model.traits.EventPayloadTrait;
+import software.amazon.smithy.aws.traits.protocols.AwsQueryCompatibleTrait;
```
Comment (syall) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java:46]: nit: added imports (both the static `writeAwsQueryErrorCodeDeserializer` and `AwsQueryCompatibleTrait` imports) don't seem to be in alphabetical order.

**Review by syall — CHANGES_REQUESTED** [review]

**Comment by eddy-aws** [comment]
Closed in favor of https://github.com/aws/aws-sdk-go-v2/pull/1934
